packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -343,19 +255,133 @@ where
         tracing::debug!("Recv messages: #{}", recv_msgs.len());
         tracing::debug!("Ack messages: #{}", ack_msgs.len());
 
-        cosmos::inject_ethereum_proofs(
-            &mut recv_msgs,
-            &mut ack_msgs,
-            &mut timeout_msgs,
-            &self.eth_client,
-            &ethereum_client_state.ibc_contract_address.to_string(),
-            ethereum_client_state.ibc_commitment_slot,
-            target_block_number,
-            trusted_slot,
-        )
-        .await?;
+        let update_msgs = if self.mock {
```

Comment:
I don't love having all of this in a big if statement.

I actually don't like having the mock handling in this file at all. I think a cleaner option would be to have a separate mock tx_builder.

Reply:
Agree. I'll did it now rather than later. Lmk what you think about the new refactor.